modules/run-sync-gateway/run-sync-gateway

```diff
@@ -182,7 +182,7 @@ function wait_for_couchbase_clusters {
   fi
 
   local databases
-  databases=($(cat "$config" | jq -c '.databases | .[]?'))
+  databases=($(cat "$config"| sed '/"import_filter": `/,/`/d' | sed '/"sync": `/,/`/d' | sed 's/"num_index_replicas": 0,/"num_index_replicas": 0/g' | jq -c '.databases | .[]?'))
```

Comment:
Does this handle multiple lines? That is, isn't the point of backticks to support multi-line scripts? So I'm trying to understand if this `sed` command will remove these multiline blocks, but still leave valid JSON behind for `jq`?

Reply:
Correct, it is **removing** the lines for the backticks because they aren't needed for the purpose of waiting on the bucket defined in the database. The original sting is not mutated and the config loads properly after the buckets are found and available.

Reply:
Oh, man, I totally misread the `sed` syntax there and didn't catch on to the `/d`. So if I understand correctly, you're using `sed`'s ability to delete lines, and the syntax there is something like:

```bash
sed '/START/,END/d'
```

Where `START` is the start of a pattern to match, `END` is the end of the pattern to match, and `/d` means "delete everything in between." Is that right? If so, a few requests:

1. Please extract this logic into a named function. Something like `remove_matching_lines`.
2. Add a comment to that function explaining this usage of `sed`, as it is _not_ the common pattern most people are used to.
3. Make sure to handle varying amounts of whitespace after the keys: e.g., between `"import_filter"` and the backtick could be 0, 1, or N spaces, tabs, etc.
4. Instead of hard-coding the keys that may include backticks, should we get rid of _all_ backticks? That is, use a pattern like:

    ```
    sed '/"(.+?)": `/,/`/d'
    ```

Reply:
Yes, you could do that, however, if a sync function is 3400 lines of code then deleting those lines makes the search for the databases and their buckets faster. You have to keep in mind what this function is doing. It's just waiting for the buckets to become available. we are only removing these lines in a disposable local variable.

Reply:
The suggestions I made above are to do _exactly_ the same thing you proposed, but with a couple minor tweaks to make it easier to understand for future maintainers, and to make the code slightly less brittle. It's still using `sed` to strip out backticks, whether they contain 4 lines or 3,400 lines of code within them.